src/main/java/ru/smirnovv/MaxBoundedTreeSet.java

```diff
@@ -5,17 +5,44 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
+/**
+ * MaxBoundedTreeSet extends the <tt>TreeSet</tt> class.
+ * This class is limited TreeSet, when elements are sorted in
+ * descending order.
+ *
+ * @see TreeSet
+ */
 public class MaxBoundedTreeSet extends TreeSet<Long> {
+    /**
+     * The maximum number of elements in the MaxBoundedTreeSet.
+     */
     private final long maxSize;
 
+    /**
+     * Constructs an empty MaxBoundedTreeSet with the specified
+     * maximum number of elements.
+     *
+     * @param maxSize the maximum number of elements
+     * @throws IllegalArgumentException if the specified maximum number
+     *                                  of elements is negative or zero
+     */
     public MaxBoundedTreeSet(/*@Nonnegative*/ final int maxSize) {
         checkArgument(maxSize > 0, "Argument 'maxSize' must be positive");
 
         this.maxSize = maxSize;
     }
 
+    /**
+     * Adds the specified element to this set if this element greater
```

Comment:
Adds a new number to this set if it's not full yet or if the given number is greater than the smallest one in it. When this set is full already and the given number added, it removes the smallest one to keep the maximum size.